### ISSUE TITLE
Preferred coding style w.r.t. tabs vs. spaces

### ISSUE THREAD

**Issue by avdgrinten** [networkit/networkit#173]
NetworKit's C++ coding style is quite inconsistent. While I will not fight a religious war about tabs vs. spaces, I  would prefer to only have to deal with one style of indentation. I am not suggesting that we should make everything consistent right now (large commits that only change whitespace are just a waste of time), but maybe we can at least come up with a guideline for new code. Existing code can be refactored when it is touched for other reasons.

Thoughts, anyone?

Here are some statistics:
```
[~/projects/networkit-main]$ grep -rlP '^\t' networkit/cpp/ | wc -l
599
[~/projects/networkit-main]$ grep -rlP '^  ' networkit/cpp/ | wc -l
138
```
i.e. wie have 600 files with tabs, only 140 with spaces.

Personally, I would vote to follow those statistics and just use tabs everywhere. Does anyone has a strong opinion for the other alternative?

**Comment by rbange** [comment]
An argument towards tab using would be that following the [devguide](https://github.com/kit-parco/networkit/blob/Dev/Doc/DevGuide.mdown) the preferred indent in Python is tab too.

>  - In Python, indent using tabs, not spaces.

A statement towards tabs in cpp would result in more simple editor configuration.

**Comment by manpen** [comment]
Since I needed to change systems/editors a couple of time in the past few weeks, I found how much I miss the automatic formatting we are using in [STXXL][(https://github.com/bingmann/stxxl). Here, we are using [Uncrustify](https://github.com/uncrustify/uncrustify) to enforce a consistent formatting of code. The tool automatically adds and removes spaces and line breaks according to a very flexible set of rules. 

While I initially considered this very invasive, I learnt to love this tool. Independent of your editor/IDE, you simple write code and do not need to think too much about your coding style; simply run the tool before any PR and be done with it. The coding style of different authors then typically looks more consistent.  It also makes diffs on GitHub much clearer, as there won't be patches changing the indentation anymore.

**Comment by hmeyerhenke** [comment]
Sounds like a good thing. When the set of contributors grows beyond a small margin, tool support is very likely what you need to ensure consistency.

**Comment by manpen** [comment]
I saw several commits by @angriman that apply `Clang Format` and think it's great. Can we have the configuration in the repo, s.t. we're using a common style at least for new contributions?

**Comment by manpen** [comment]
Solved in PRs #411 #412 #430
